scripts/check_build.sh

```diff
@@ -0,0 +1,45 @@
+#!/bin/bash
+
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -euo pipefail
+
+# Confirm we have run `make build` by:
+# 1) Running make build in a tmp directory
```

Comment:
I think we can leverage git and run "make build" then "git status --short" to see if files are changed (it will print output if changes exist, nothing if no changes).

Reply:
That would work fine in CI, but I also want this to be runnable locally (where we don't want to run `make build` on the live git install).

Reply:
I'm not quite sure I understand the local case, what's the user/developer flow look like there?

Reply:
For things like git precommit checks it's helpful to be able to run simple linters like this locally.

It's admittedly not an especially common case, but I don't think it's worth changing to something else given this.